src/components/cylc/tree/Tree.vue

```diff
@@ -168,80 +119,45 @@ export default {
       cyclePointsOrderDesc = JSON.parse(localStorage.cyclePointsOrderDesc)
     }
     this.cyclePointsOrderDesc = cyclePointsOrderDesc
+
+    // Reactively run filtering
+    if (this.nodeFilterFunc) {
+      this.$watch(
+        () => [this.filterState, this.rootChildren],
+        ([active, nodes], [wasActive, oldNodes]) => {
+          if (active) {
+            // Need to wipe cache due to mem leak bug in Vue < 3.4
+            // https://github.com/vuejs/core/issues/9419
+            // TODO: can remove this line after upgrade to Vue 3.4:
+            this.filteredOutNodesCache = new WeakMap()
```

Comment:
vuejs/core#9419

I tested out tree view filtering without this line and found a memory leak with our current Vue version 3.3

I then tested with Vue 3.4.0-alpha.1 and did not get a memory leak

Reply:
Could you put up an issue for this and stick it into the comment to ensure we get round to it.

Reply:
I'm not sure if it's important enough to track in its own issue - creating a new map can't have that much of a performance penalty, can it?

Reply:
No, an issue for the TODO!

Reply:
That's what I'm saying - if we forget to act on this TODO it won't matter too much. But if someone comes across the TODO after the Vue version has been bumped they can address it. I may even get rid of this soon if I can get the computed properties approach working.